CMakeLists.txt

```diff
@@ -126,6 +126,10 @@ if (PUBLIC_DEPS)
   target_link_libraries(kokkosremote PUBLIC ${PUBLIC_DEPS})
 endif()
 
+if (KRS_ENABLE_NVSHMEMSPACE)
+  target_link_libraries(kokkosremote PRIVATE "-lnvidia-ml")
+endif()
```

Comment:
Don't you need to use `PUBLIC` if you want to export the flag: https://cmake.org/cmake/help/latest/command/target_link_libraries.html#libraries-for-a-target-and-or-its-dependents

Reply:
Using libkokkosremote.a in a downstream project, CMAKE seems to include the private link lib of that library. I am seeing the correct invocation. 

A CmakeLists text could look like this:
```cmake
cmake_minimum_required(VERSION 3.13)
project(Example LANGUAGES CXX)
find_package(KokkosRemote REQUIRED)
file(GLOB SOURCES *.cpp)
foreach(SRC_FILE ${SOURCES})
  message(STATUS "Adding source file:" ${SRC_FILE})
  get_filename_component(SRC_FILE_NAME ${SRC_FILE} NAME)
  string(REGEX REPLACE "\\.[^.]*$" "" SRC_FILE_NAME ${SRC_FILE_NAME})
  add_executable(${SRC_FILE_NAME} ${SRC_FILE})
  target_link_libraries(${SRC_FILE_NAME} Kokkos::kokkosremote)
endforeach()
```

The link phase of the test app is then:
....o/software/nvshmem/lib/libnvshmem.a -lnvidia-ml